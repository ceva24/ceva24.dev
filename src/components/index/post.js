import React from "react"
import { css } from "@emotion/core"
import { rhythm } from "src/utils/typography"
import {Link} from "gatsby";

export default ({node}) => (
    <div key={node.id}>
        <h3 css={css`
            margin-bottom: ${rhythm(1 / 4)};
        `}>
            <Link to={node.fields.path} css={css`
                text-decoration: none;
            `}>
                {node.frontmatter.title}{" "}
            </Link>
            <span css={css`
                color: #7f8c8d;
            `}>
                â€” {node.frontmatter.date}
            </span>
        </h3>
        <p>{node.excerpt}</p>
    </div>
)
