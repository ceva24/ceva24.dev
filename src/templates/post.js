/* eslint-disable react/no-danger */
import React from "react";
import { graphql, Link } from "gatsby";
import { css } from "@emotion/react";
import { rhythm } from "src/utils/typography";
import Layout from "src/components/layout";
import { PostDate } from "src/components/post-date";

const Post = ({ data }) => (
    <Layout title={data.markdownRemark.frontmatter.title}>
        <h2
            css={css`
                margin-bottom: 0;
            `}
        >
            {data.markdownRemark.frontmatter.title}
        </h2>
        <div
            css={css`
                margin-bottom: ${rhythm(3 / 4)};
            `}
        >
            <PostDate>{data.markdownRemark.frontmatter.date}</PostDate>
        </div>
        <div dangerouslySetInnerHTML={{ __html: data.markdownRemark.html }} />
        <Link to="/">Home</Link>
    </Layout>
);

export default Post;

export const query = graphql`
    query ($slug: String!) {
        markdownRemark(fields: { slug: { eq: $slug } }) {
            html
            frontmatter {
                title
                date
            }
        }
    }
`;
