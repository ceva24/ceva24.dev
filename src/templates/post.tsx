import React from "react";
import { graphql, Link } from "gatsby";
import { Layout } from "../components/layout";

const PurePost: React.FC<PostTemplateData> = ({ data }: PostTemplateData) => (
    <>
        <article>
            <div className="mt-10 mb-5 space-y-3">
                <h2 className="text-4xl uppercase font-normal">
                    {data.markdownRemark.frontmatter.title}
                </h2>

                <div>{data.markdownRemark.frontmatter.date}</div>
            </div>

            <div
                // eslint-disable-next-line react/no-danger
                dangerouslySetInnerHTML={{ __html: data.markdownRemark.html }}
            />
        </article>
        <Link to="/">Home</Link>
    </>
);

const Post: React.FC<PostTemplateData> = (data: PostTemplateData) => (
    <Layout title={data.data.markdownRemark.frontmatter.title}>
        <PurePost {...data} />
    </Layout>
);

// eslint-disable-next-line @typescript-eslint/no-confusing-void-expression
const query = graphql`
    query ($slug: String!) {
        markdownRemark(fields: { slug: { eq: $slug } }) {
            html
            frontmatter {
                title
                date
            }
        }
    }
`;

export default Post;
export { PurePost, query };
