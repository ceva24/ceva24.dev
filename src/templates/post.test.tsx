import React from "react";
import { render } from "@testing-library/react";
import { Head, PurePost } from "./post";

jest.mock("gatsby", (): unknown => {
    return {
        ...jest.requireActual("gatsby"),
        graphql: jest.fn(),
        StaticQuery: jest.fn(),
        useStaticQuery: jest.fn(),
    };
});

jest.mock("@gatsbyjs/reach-router", (): unknown => ({
    ...jest.requireActual("@gatsbyjs/reach-router"),
    useLocation: jest.fn().mockImplementation(() => {
        return { pathname: "/" };
    }),
}));

const data: PostTemplateData = {
    data: {
        markdownRemark: {
            frontmatter: {
                title: "Post title",
                date: "2000-01-01",
            },
            html: "<p>Hello world</p>",
        },
    },
};

describe("post", () => {
    it("renders", () => {
        const { asFragment } = render(<PurePost {...data} />);

        expect(asFragment()).toMatchSnapshot();
    });
});

describe("head", () => {
    it("renders", () => {
        const { asFragment } = render(<Head {...data} />);

        expect(asFragment()).toMatchSnapshot();
    });
});
