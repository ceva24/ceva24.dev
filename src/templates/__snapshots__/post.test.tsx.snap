// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`post renders 1`] = `
<DocumentFragment>
  <article>
    <div
      class="mt-10 mb-5 space-y-3"
    >
      <h2
        class="text-4xl uppercase font-normal"
      >
        Post title
      </h2>
      <div>
        2000-01-01
      </div>
    </div>
    <div>
      <p>
        Hello world
      </p>
    </div>
  </article>
  <div
    class="min-w-max text-center"
  >
    <a
      aria-current="page"
      class=""
      href="/"
    >
      Home
    </a>
  </div>
</DocumentFragment>
`;
